internal/tests/cli/boundary/target.bats

```diff
@@ -100,3 +100,36 @@ load _helpers
   run read_target $id
   [ "$status" -eq 1 ]
 }
+
+@test "boundary/target: create target with a network address" {
+  run create_tcp_target_with_addr $DEFAULT_P_ID "localhost" 22 $TGT_NAME_WITH_ADDR
+  [ "$status" -eq 0 ]
+}
+
+@test "boundary/target: cannot assign an host source to a target with an address" {
+  local id=$(target_id_from_name $DEFAULT_P_ID $TGT_NAME_WITH_ADDR)
+  run assoc_host_sources $id $DEFAULT_HOST_SET
+  [ "$status" -eq 1 ]
+}
+
+@test "boundary/target: can assign an host source to a target after deleting address" {
+  local id=$(target_id_from_name $DEFAULT_P_ID $TGT_NAME_WITH_ADDR)
+  run update_address $id "null"
+  [ "$status" -eq 0 ]
+  run assoc_host_sources $id $DEFAULT_HOST_SET
+  [ "$status" -eq 0 ]
+}
+
+@test "boundary/target: cannot assign an address to a target with an host source" {
+  local id=$(target_id_from_name $DEFAULT_P_ID $TGT_NAME_WITH_ADDR)
+  run update_address $id "localhost"
+  [ "$status" -eq 1 ]
+}
+
+@test "boundary/target: can assign an an address to a target after deleting host source" {
+  local id=$(target_id_from_name $DEFAULT_P_ID $TGT_NAME_WITH_ADDR)
+  run remove_host_sources $id $DEFAULT_HOST_SET
+  [ "$status" -eq 0 ]
+  run update_address $id "localhost"
+  [ "$status" -eq 0 ]
+}
```

Comment:
Can we add a test that tries to unset an address by setting it to "null", as we have a handler for in the CLI? And that the host source association can then be set.

Reply:
The bats test does validate this behavior. Because the bats tests are not isolated from each other, the state changes from one test carries on to the next. The following tests does what you suggest:
```
@test "boundary/target: create target with a network address" 
@test "boundary/target: cannot assign an host source to a target with an address"
@test "boundary/target: can assign an host source to a target after deleting address"
@test "boundary/target: cannot assign an address to a target with an host source"
@test "boundary/target: can assign an an address to a target after deleting host source"
```

Reply:
Cool, thanks!